data-prepper-logstash-configuration/src/main/java/org/opensearch/dataprepper/logstash/LogstashConfigConverter.java

```diff
@@ -0,0 +1,59 @@
+package org.opensearch.dataprepper.logstash;
+
+import com.amazon.dataprepper.model.configuration.PipelineModel;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.fasterxml.jackson.dataformat.yaml.YAMLGenerator;
+import org.antlr.v4.runtime.CharStreams;
+import org.antlr.v4.runtime.CommonTokenStream;
+import org.antlr.v4.runtime.tree.ParseTree;
+import org.opensearch.dataprepper.logstash.mapping.LogstashMapper;
+import org.opensearch.dataprepper.logstash.model.LogstashConfiguration;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.Map;
+
+/**
+ * Converts Logstash configuration file and returns YAML file location
+ *
+ * @since 1.2
+ */
+public class LogstashConfigConverter {
+    public String convertLogstashConfigurationToPipeline(String logstashConfigurationPath, String outputDirectory) throws IOException {
+        Path configurationFilePath = Paths.get(logstashConfigurationPath);
+        final String logstashConfigAsString = new String(Files.readAllBytes(configurationFilePath));
+
+        LogstashLexer lexer = new LogstashLexer(CharStreams.fromString(logstashConfigAsString));
+        CommonTokenStream tokens = new CommonTokenStream(lexer);
+        LogstashParser parser = new LogstashParser(tokens);
+        ParseTree tree = parser.config();
+
+        org.opensearch.dataprepper.logstash.parser.LogstashVisitor visitor = new org.opensearch.dataprepper.logstash.parser.LogstashVisitor();
+        LogstashConfiguration logstashConfiguration = (LogstashConfiguration) visitor.visit(tree);
+
+        LogstashMapper logstashMapper = new LogstashMapper();
+        PipelineModel pipelineModel = logstashMapper.mapPipeline(logstashConfiguration);
+        Map<String, Object> pipeline = Collections.singletonMap("logstash-converted-pipeline", pipelineModel);
+
+        ObjectMapper mapper = new ObjectMapper(YAMLFactory.builder()
+                .disable(YAMLGenerator.Feature.WRITE_DOC_START_MARKER)
+                .enable(YAMLGenerator.Feature.INDENT_ARRAYS_WITH_INDICATOR)
+                .disable(YAMLGenerator.Feature.SPLIT_LINES)
+                .enable(YAMLGenerator.Feature.LITERAL_BLOCK_STYLE)
+                .build());
+
+        String confFileName = configurationFilePath.getFileName().toString();
```

Comment:
Do we have all the fields that should not be changed `final`?

Reply:
No, will update it